### PR TITLE
Background tasks

### PR DESCRIPTION
**Pull request by JFer11**
In this branch background tasks were added, as well as, a feature that send a mail to recover your password.

In addition, there was a big change on the way the app initialize itself. Was created more modules to improve readability. Taking into account how modules are imported and what they do, some modules were refactored.
Also change the way the app configure itself (now is from an object that contains several configuration)
Admin feature was moved from "app_init" to an individual module.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 *.pyc
 .env
 .idea
+app_init_backUp.py
+database_config.py
```
Comment (joaquinco) [inline .gitignore]: missing new line.

Generally you don't want to ignore every possible file you might have in your local environment but just files that are generated by running the project or configuration files that contain secrets.

For this kind you might want to ignore them or just delete them.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
  To use an emulated email server, Python provides one that is very handy that you can start in a
- second terminal with the following command: "$ python -m smtpd -n -c DebuggingServer localhost:8025"
- 
\ No newline at end of file
+ second terminal with the following command:
+ 
+ ```"$ python -m smtpd -n -c DebuggingServer localhost:8025"```
```
Comment (joaquinco) [inline README.md]: Double quotes are not needed here

File: training/background_tasks/tasks.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    job = get_current_job()
+    print('Starting task')
+    for i in range(seconds):
+        job.meta['progress'] = 100.0 * i / seconds
+        job.save_meta()
+        print(i)
+        time.sleep(1)
+    job.meta['progress'] = 100
+    job.save_meta()
+    print('Task completed')
```
Comment (joaquinco) [inline training/background_tasks/tasks.py]: missing new line.

Shouldn't email sending tasks be here?

File: training/config.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import os
+
+
+# default config
+class BaseConfig(object):
+    DEBUG = False
+    # shortened for readability
+    #SECRET_KEY = '\xbf\xb0\x11\xb1\xcd\xf9\xba\x8bp\x0c...'
```
Comment (joaquinco) [inline training/config.py]: Commented code. Maybe it's worth leaving the SECRET_KEY attribute and read its value from env?

File: training/config.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    WTF_CSRF_ENABLED = False
+    SQLALCHEMY_DATABASE_URI = os.environ['DATABASE_URL_FOR_TEST']
+
+
+class DevelopmentConfig(BaseConfig):
+    DEBUG = True
+
+
+class ProductionConfig(BaseConfig):
+    DEBUG = False
```
Comment (joaquinco) [inline training/config.py]: Missing new line

File: training/controllers/admin/admin_web.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+from flask_admin import Admin
+from flask_admin.contrib.sqla import ModelView
+from training.models.users import User
+from training.app import db, app, bcrypt
+
+admin = Admin(app, name='Users admin', template_mode='bootstrap3')
+
+
+class MyModelView(ModelView):
```
Comment (joaquinco) [inline training/controllers/admin/admin_web.py]: This class should have the name of the model, like `UserModelView` or something like that

File: training/main.py [hunk 7]
```diff
@@ -1,22 +1,22 @@
...
 
-def init_db():
+#def init_db():
     #global engine
     #global session
-    from training.models import users
+   # from training.models import users
     #Base.metadata.drop_all(engine)
-    Base.metadata.create_all(engine)
+#    Base.metadata.create_all(engine)
```
Comment (joaquinco) [inline training/main.py]: Should this file be deleted?

File: training/models/users.py [hunk 8]
```diff
@@ -1,16 +1,15 @@
...
-	password = Column(String(1000), nullable=False)
-	email = Column(String(50), nullable=False, unique=True)
-	mail_validation = Column(Boolean, unique=False, default=False)
-	created_at = Column(DateTime(), default=datetime.now())
+	id = db.Column(db.String(50), primary_key=True)
+	username = db.Column(db.String(50), nullable=False, unique=True)
+	password = db.Column(db.String(1000), nullable=False)
+	email = db.Column(db.String(50), nullable=False, unique=True)
+	mail_validation = db.Column(db.Boolean, unique=False, default=False)
+	created_at = db.Column(db.DateTime(), default=datetime.now())
```
Comment (joaquinco) [inline training/models/users.py:12]: What's the difference of using these fields from `db` or `sqlalchemy`?

File: training/templates/email_form_template.html [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+	    {{ render_field(form.email_recipient) }}
+        {{ render_field(form.message_body) }}
+        {{ render_field(form.send_now) }}
+	    <input type="submit" name="boton">
+  		</dl>
+
+		{%if sended %}
+		<p>Email enviado!</p>
+		{%endif%}
+	</form>
```
Comment (joaquinco) [inline training/templates/email_form_template.html]: Indentation and spacing is weird

**Review by joaquinco — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (dcousin24) [reply README.md]: Yeah, and for one line comments you can use # also if you want

File: training/templates/email_form_template.html (hunk 9 above)
Comment (dcousin24) [reply training/templates/email_form_template.html]: Yeah, have in mind that probably you will have to come back to this code and it's important that it's easy to read.

**Comment by joaquinco** [comment]
Please add missing new line and remove commented code and we're good to go

**Review by joaquinco — APPROVED** [review]

**Pull request merged**